src/leveled_penciller.erl

```diff
@@ -809,21 +795,16 @@ handle_call({fetch_keys,
     
     %% Rename any reference to loop state that may be used by the function
     %% to be returned - https://github.com/martinsumner/leveled/issues/326
-    Manifest = State#state.manifest,
+    SSTiter =
+        case State#state.query_manifest of
+            undefined ->
+                leveled_pmanifest:query_manifest(
+                    State#state.manifest, StartKey, EndKey);
+            QueryManifest ->
+                QueryManifest
```

Comment:
I have a bit of problem to see whether this is compatible with the old solution.

If you register a snapshot, then you now put the query_manifest in the state (line 881). A consecutive fetch_keys will use that query.
But the StartKey and EndKey of the registers snapshot and the fetch might be different? In which case the result of the fetch is different??

More precede, if the fetch is on Query {StartKey = 10, EndKey = 20} and the fetch is on StartKey = 1, EndKey = 8, then the original would use the 1..8 range for fetch, whereas the new one would have saved the query 10...20 in the state and use that. What am I missing?

Reply:
If at the point of the snapshot you know the Query, the the StartKey/EndKey are provided here.  So it is assumed that there will only be subsequent call to find_keys on that snapshot, and it will be for the same StartKey/EndKey.

There are circumstances where a snapshot is reused for multiple queries (e.g. when doing list buckets), but in this case there is no Query provided when registering the snapshot.

I think it might be worth making the query_manifest a tuple of {SubManifest, StartKey, EndKey} and a guard to crash a subsequent find_keys request which tries a StartKey or EndKey outside of the range.  This will then make this requirement explicit.